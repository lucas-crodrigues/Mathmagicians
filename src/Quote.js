import Header from './components/Header';
import Footer from './components/Footer';
import './Page.css';

const Quote = () => (
  <div className="quote">
    <Header />
    <h3
      style={{
        margin: '4vh 2vw',
        fontSize: '40px',
        paddingBottom: '30vh',
      }}
    >
      “Sometimes the questions are complicated and the answers are simple.” – Dr. Seuss
    </h3>
    <Footer />
  </div>
);

export default Quote;
