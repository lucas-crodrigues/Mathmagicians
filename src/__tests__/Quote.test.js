import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

describe('Quote page test suites', () => {
  test('Quote page snapshot', () => {
    const renderedComponent = renderer.create(<BrowserRouter><Quote /></BrowserRouter>).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('Check quote presencce', () => {
    render(<BrowserRouter><Quote /></BrowserRouter>);
    const quote = screen.getByRole('heading', { name: /“sometimes the questions are complicated and the answers are simple\.” – dr\. seuss/i });
    expect(quote).toHaveTextContent('“Sometimes the questions are complicated and the answers are simple.” – Dr. Seuss');
  });
});
