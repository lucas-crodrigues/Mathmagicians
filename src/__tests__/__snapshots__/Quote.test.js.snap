// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote page test suites Quote page snapshot 1`] = `
<div
  className="quote"
>
  <header
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottom": "1px solid gray",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "4vh 2vw",
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      Math Magicians
    </h1>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-around",
          "width": "25vw",
        }
      }
    >
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </nav>
    </div>
  </header>
  <h3
    style={
      Object {
        "fontSize": "40px",
        "margin": "4vh 2vw",
        "paddingBottom": "30vh",
      }
    }
  >
    “Sometimes the questions are complicated and the answers are simple.” – Dr. Seuss
  </h3>
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "borderTop": "1px solid black",
        "height": "25px",
      }
    }
  />
</div>
`;
